File: Haskell/euler002/test/Euler002Test.hs
import Test.Tasty
import Test.Tasty.HUnit

import Euler002

unitTests :: TestTree
unitTests = testGroup "Euler - 002: Unit Tests"
  [ testCase
      "The first ten Fibonacci numbers are \
      \1, 2, 3, 5, 8, 13, 21, 34, 55 and 89" $
      take 10 fibs @?=
      [1, 2, 3, 5, 8, 13, 21, 34, 55, 89]

  , testCase
      "Final Answer: The sum of the even-valued Fibonacci numbers whose values \
      \do not exceed four million is 4613732" $
      euler002 @?= 4613732
  ]

main :: IO()
main = defaultMain unitTests

File: CXX/tests/euler002_test.cpp
/*
 * Project Euler - 002 - Even Fibonacci Numbers (Test)
 * Tamara Roberson <tamara.roberson@gmail.com>
 * Copyright (c) 2019 Tamara Roberson
 */

#include <euler/euler002.hpp>

#include <catch2/catch.hpp>

TEST_CASE("Euler 002: Example is correct")
{
    vector<int> fibs = { 1, 2, 3, 5, 8, 13, 21, 34, 55, 89 };
    REQUIRE(euler002_example() == fibs);
}

TEST_CASE("Euler 002: Final answer is correct")
{
    REQUIRE(euler002() == 4613732);
}
